packages/angular-cli/models/webpack-build-styles.ts

```diff
@@ -0,0 +1,130 @@
+  target: string,
+  sourcemap: boolean,
+  outputHashing: string,
+  extractCss: boolean,
+) {
+
+  const appRoot = path.resolve(projectRoot, appConfig.root);
+  const entryPoints: { [key: string]: string[] } = {};
+  const globalStylePaths: string[] = [];
+  const extraPlugins: any[] = [];
+
+  // discard comments in production
+  const extraPostCssPlugins = target === 'production' ? [postcssDiscardComments] : [];
+
+  // determine hashing format
+  const hashFormat = getOutputHashFormat(outputHashing);
+
+  // use includePaths from appConfig
+  const includePaths: string [] = [];
+
+  if (appConfig.stylePreprocessorOptions
+    && appConfig.stylePreprocessorOptions.includePaths
+    && appConfig.stylePreprocessorOptions.includePaths.length > 0
+  ) {
+    appConfig.stylePreprocessorOptions.includePaths.forEach((includePath: string) =>
+      includePaths.push(path.resolve(appRoot, includePath)));
+  }
+
+  // process global styles
+  if (appConfig.styles.length > 0) {
+    const globalStyles = extraEntryParser(appConfig.styles, appRoot, 'styles');
+    // add style entry points
+    globalStyles.forEach(style =>
+      entryPoints[style.entry]
+        ? entryPoints[style.entry].push(style.path)
+        : entryPoints[style.entry] = [style.path]
+    );
+    // add global css paths
+    globalStylePaths.push(...globalStyles.map((style) => style.path));
+  }
+
+  // set base rules to derive final rules from
+  const baseRules = [
+    { test: /\.css$/, loaders: [] },
+    { test: /\.scss$|\.sass$/, loaders: ['sass-loader'] },
+    { test: /\.less$/, loaders: ['less-loader'] },
+    // stylus-loader doesn't support webpack.LoaderOptionsPlugin properly,
+    // so we need to add options in it's query
+    { test: /\.styl$/, loaders: [`stylus-loader?${JSON.stringify({
+      sourceMap: sourcemap,
+      paths: includePaths
+    })}`] }
+  ];
+
+  const commonLoaders = ['postcss-loader'];
+
+  // load component css as raw strings
+  let rules: any = baseRules.map(({test, loaders}) => ({
+    exclude: globalStylePaths, test, loaders: ['raw-loader', ...commonLoaders, ...loaders]
+  }));
+
+  // load global css as css files
+  if (globalStylePaths.length > 0) {
+    rules.push(...baseRules.map(({test, loaders}) => ({
+      include: globalStylePaths, test, loaders: ExtractTextPlugin.extract({
+        remove: false,
+        loader: ['css-loader', ...commonLoaders, ...loaders],
+        fallbackLoader: 'style-loader'
+      })
+    })));
+  }
+
+  // supress empty .js files in css only entry points
+  if (extractCss) { extraPlugins.push(new SuppressExtractedTextChunksWebpackPlugin()); }
+
+  return {
+    entry: entryPoints,
+    module: { rules },
+    plugins: [
+      // extract global css from js files into own css file
+      new ExtractTextPlugin({
+        filename: `[name]${hashFormat.extract}.bundle.css`,
+        disable: !extractCss
+      }),
+      new webpack.LoaderOptionsPlugin({
+        test: /\.(css|scss|sass|less|styl)$/,
```

Comment:
This test technically isn't needed and It adds an additional check per resource request.